frontend/src/app/messages/messages.service.spec.ts

```diff
@@ -0,0 +1,171 @@
+import { fakeAsync, tick, TestBed, inject } from '@angular/core/testing';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { Observable } from 'rxjs/Observable';
+import { ISubscription } from 'rxjs/Subscription';
+import { environment } from '../../environments/environment';
+
+
+import {
+  MessagesService, Message,
+  MessageType, DEFAULT_TIMEOUT_TIME, ServerMessage
+} from './messages.service';
+
+describe('MessagesService', () => {
+  let service: MessagesService;
+  let httpMock: HttpTestingController;
+  let testMessage: Message;
+  let numberOfCalls: number;
+  let messageSubscription: ISubscription;
+
+  this.subscribeToMessage$ = () => {
+    messageSubscription = service.message$.subscribe(msg => {
+      expect(msg).toEqual(testMessage)
+      numberOfCalls++;
+    })
+  }
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [MessagesService]
+    });
+
+    service = TestBed.get(MessagesService);
+    httpMock = TestBed.get(HttpTestingController);
+    testMessage = {
+      message: 'Test message',
+      type: MessageType.Info,
+      dismissible: true,
+      timeout: DEFAULT_TIMEOUT_TIME
+    }
+    numberOfCalls = 0;
+    this.subscribeToMessage$();
+  });
+
+  afterEach(() => {
+    httpMock.verify();
+    service = null;
+    testMessage = null;
+
+    if (messageSubscription) {
+      messageSubscription.unsubscribe();
+      messageSubscription = null;
+    }
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  describe('Messages are correctly exposed through observable', () => {
+    it('showMessage() should send generic message', fakeAsync(() => {
+      // Method called with one parameter
+      service.showMessage(testMessage.message);
+
+      // Method called with multiple parameters
+      testMessage.type = MessageType.Danger;
+      testMessage.dismissible = false;
+      testMessage.timeout = 5;
+      service.showMessage(testMessage.message, MessageType.Danger, false, 5);
+
+      expect(numberOfCalls).toEqual(2);
```

Comment:
This test tests nothing at first glance. I'm not a big fan of hiding expects in outside function. It seems detached from context. Even if there is slight code repetition it improves readability a lot.

Reply:
Done